lile.go

```diff
@@ -68,47 +110,174 @@ func AddStreamInterceptor(sint grpc.StreamServerInterceptor) {
 	service.StreamInts = append(service.StreamInts, sint)
 }
 
-func Server(r registerImplementation) {
-	service.GRPCImplementation = r
+func Run() {
+	// add to registry
+	if service.Config.UsesRegistry() {
+		rc, rcErr := GetRegistryClient()
+		if rcErr == nil {
+			err := rc.Register(service.ID, service.Name, service.Config.Service.Port, &api.AgentServiceCheck{
+				CheckID:                        "",
```

Comment:
We could abastract this out and just pass the `lile` service type or similar.

Also by default in Go, everything is it's empty type, so you don't need to the fill the struct in 😸

Reply:
I don't understand what you mean? Hide it in another file? :-) 

I know about the default values, I just wanted to leave it there for now, so I can play around with it. This struct cannot stay as-is, because of consul's abstraction. :)